examples/train_model.py

```diff
@@ -70,9 +70,8 @@ def main():
     task = EnergyReconstruction(
         hidden_size=gnn.nb_outputs,
         target_label=target,
-        loss_function=LogCoshLoss(
```

Comment:
My immediate thought is that it might be a bit confusing for both the `LossFunction` class and the `Task` class to support `transform_*` arguments, and the latter passing them on to the former. I like the idea, as proposed in this PR, to let the `Task` class handle all of the transforms. Then the `LossFunction` class is simpler, and just handles the actual computation of the loss. I am going to propose edits in this vein below, and then we can discuss if that is the path we'd like to take. If so, this PR should also update the `LossFunction` class accordingly.